llvm/lib/SandboxIR/SandboxIR.cpp

```diff
@@ -687,7 +689,10 @@ void LoadInst::dump() const {
 void StoreInst::setVolatile(bool V) {
   auto &Tracker = Ctx.getTracker();
   if (Tracker.isTracking())
```

Comment:
we might even be able to factor out `if (Tracker.isTracking())` somehow with another helper method. but maybe that's not good for consistency with the other trackers that don't use `GenericSetter`

Reply:
Yeah, something like that. The part about it conditionally creating the object internally is the tricky part.